File: opensource/data/src/main/java/com/sos/scheduler/engine/data/base/StringValue.java
package com.sos.scheduler.engine.data.base;

public abstract class StringValue {
    private final String string;

    protected StringValue(String x) {
        if (x == null)  throw new NullPointerException(getClass().toString());
        string = x;
    }

    public final String asString() {
        return string;
    }

    public final boolean isEmpty() {
        return string.isEmpty();
    }

    @Override public boolean equals(Object o) {
        return o != null && getClass() == o.getClass() && string.equals(((StringValue)o).string);
    }

    @Override public final int hashCode() {
        return string.hashCode();
    }

    @Override public String toString() { 
        return asString();
    }
}

File: opensource/data/src/main/scala/com/sos/scheduler/engine/data/base/StringValue.scala
package com.sos.scheduler.engine.data.base

abstract class StringValue(val string: String) extends IsString {
  /** @deprecated */
  final def asString = string
}

object StringValue {
  /** Für &lt;elememt attribute={stringValue}/>. */
  implicit def toXmlText(o: StringValue) = new xml.Text(o.string)
}